extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { IProjectProvider, WorkspaceTreeItem } from 'dataworkspace';
+import { IWorkspaceService } from '../common/interfaces';
+export class ProjectDashboard {
+
+	private dashboard: azdata.window.ModelViewDashboard | undefined;
+	private modelView: azdata.ModelView | undefined;
+	private projectProvider: IProjectProvider | undefined;
+	private overviewTab: azdata.DashboardTab | undefined;
+
+	constructor(private workspaceService: IWorkspaceService, private treeItem: WorkspaceTreeItem) {
+	}
+
+	public async showDashboard(): Promise<void> {
+		const project = this.treeItem.element.project;
+		const projectTypeId = project.projectFilePath;
+		this.projectProvider = await this.workspaceService.getProjectProvider(vscode.Uri.file(projectTypeId));
```

Comment:
the variable name is misleading, it is not projectTypeId, it is projectFilePath, you don't actually need to introduce this variable name, just you project.projectFilePath
`
vscode.Uri.file(projectTypeId)
`

Reply:
Yep, will remove it.